### PR TITLE
Cbrune/BFD-235 - update mapping spreadsheet to include fields for the beneficiary and inpatient claims table

### PR DESCRIPTION
**Pull request by cbrunefearless**
### Change Details

Added fields to the beneficiary and inpatientclaims table.  The following is what has been added and the criteria associated with it.

Screen Shot 2020-10-27 at 11 32 02 AM

### Acceptance Validation

All the existing tests pass, the right tables are altered as well as the right columns and data types correspond.

### Feedback Requested

Look at the thoroughness from the rif spreadsheet, to the loader, to the different sample .txt files in the tests. All the existing tests pass, the right tables are altered as well as the right columns and data types correspond.  The fields that were added to the rif spreadsheet have been added to the beneficiary and inpatient sheets.  I also added alter scripts V26 in the migration folder to alter the beneficiary table and the inpatient claims data.

### External References

- BFD 235 -Ingest new data from CCW: Update Mapping Spreadsheet (2)

### Security Implications

- [ ] new software dependencies

- [ ] altered security controls

- [ X] new data stored or transmitted
New fields being sent over for CCW.  
 
Screen Shot 2020-10-27 at 11 32 02 AM

In the Database 
   **Beneficiary Table**
"derivedMailingAddress1" varchar(40);
"derivedMailingAddress2" varchar(40);
"derivedMailingAddress3" varchar(40);
"derivedMailingAddress4" varchar(40);
"derivedMailingAddress5" varchar(40);
"derivedMailingAddress6" varchar(40);
"derivedCityName" varchar(100);
"derivedStateCode" varchar(2);
"derivedZipCode" varchar(9);
"mbiEffectiveDate" date;
"mbiObsoleteDate" date;

  **InpatientClaims Table**
"clmUncompensatedCareAmount" numeric(38, 2)
"fiDocumentClaimControlNumber" varchar(23);
"fiOriginalClaimControlNumber" varchar(23);

 **BeneficiariesHistory Table**
"mbiEffectiveDate" date;
"mbiObsoleteDate" date;

 **CarrierClaims Table**
"claimCarrierControlNumber" varchar(23);

 **DMEClaims Table**
"claimCarrierControlNumber" varchar(23);

 **HHAClaims Table**
"fiDocumentClaimControlNumber" varchar(23);
"fiOriginalClaimControlNumber" varchar(23);

 **HospiceClaims Table**
"fiDocumentClaimControlNumber" varchar(23);
"fiOriginalClaimControlNumber" varchar(23);

 **OutpatientClaims Table**
"fiDocumentClaimControlNumber" varchar(23);
"fiOriginalClaimControlNumber" varchar(23);

 **SNFClaims Table**
"fiDocumentClaimControlNumber" varchar(23);
"fiOriginalClaimControlNumber" varchar(23);

  

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

### CODE REVIEW

**Comment by cbrunefearless** [comment]
There was a correction to MBI obsolete date..
Screen Shot 2020-07-01 at 10 43 28 PM

**Comment by karlmdavis** [comment]
Heads up: since this stores new data we didn't have before (even though the API isn't yet sending it out), you're likely going to need to complete and get approval for a Security Impact Assessment before this gets merged. (Unless you already have some form of written okay from Casey and/or @StewGoin.)

**Comment by karlmdavis** [comment]
What's up with the CI build failure? That one at least _looks_ relevant to me, on a brief inspection.

File: apps/bfd-model/bfd-model-rif-samples/src/main/resources/rif-static-samples/sample-a-beneficiaries.txt
Comment (karlmdavis) [inline apps/bfd-model/bfd-model-rif-samples/src/main/resources/rif-static-samples/sample-a-beneficiaries.txt:2]: Please don't use realistic-looking data here. I know it's obvious to you that it's just your home address (or whatever it is), but you're setting up some future developer or security person to have a panic attack.

For addresses, I like "123 MAIN ST, APT 456, PODUNK, IA, 12345" or something along those lines. Visually, it's obvious to anyone that it didn't come from real data.

File: apps/bfd-model/bfd-model-rif-samples/src/main/resources/rif-static-samples/sample-hicn-mult-bene-beneficiaries.txt
Comment (karlmdavis) [inline apps/bfd-model/bfd-model-rif-samples/src/main/resources/rif-static-samples/sample-hicn-mult-bene-beneficiaries.txt:11]: Same as above: lean more towards non-realistic data.

File: apps/bfd-model/bfd-model-rif-samples/src/main/resources/rif-static-samples/sample-mct-beneficiaries.txt
Comment (karlmdavis) [inline apps/bfd-model/bfd-model-rif-samples/src/main/resources/rif-static-samples/sample-mct-beneficiaries.txt:9]: Same as above: lean more towards non-realistic data.

File: apps/bfd-model/bfd-model-rif-samples/src/main/resources/rif-static-samples/sample-mct-manifest.xml
Comment (karlmdavis) [inline apps/bfd-model/bfd-model-rif-samples/src/main/resources/rif-static-samples/sample-mct-manifest.xml:2]: WTG!

File: apps/bfd-model/bfd-model-rif-samples/src/main/resources/rif-static-samples/sample-mct-update-1-beneficiaries.txt
Comment (karlmdavis) [inline apps/bfd-model/bfd-model-rif-samples/src/main/resources/rif-static-samples/sample-mct-update-1-beneficiaries.txt:9]: Same as above: lean more towards non-realistic data.

File: apps/bfd-model/bfd-model-rif-samples/src/main/resources/rif-static-samples/sample-mct-update-4-beneficiaries.txt
Comment (karlmdavis) [inline apps/bfd-model/bfd-model-rif-samples/src/main/resources/rif-static-samples/sample-mct-update-4-beneficiaries.txt:11]: Same as above: lean more towards non-realistic data.

File: apps/bfd-model/bfd-model-rif-samples/src/main/resources/rif-static-samples/sample-mct-update-pde-5-manifest.xml
Comment (karlmdavis) [inline apps/bfd-model/bfd-model-rif-samples/src/main/resources/rif-static-samples/sample-mct-update-pde-5-manifest.xml:2]: Is the schema version here correct?

File: apps/bfd-model/bfd-model-rif-samples/src/main/resources/rif-static-samples/sample-u-beneficiaries.txt
Comment (karlmdavis) [inline apps/bfd-model/bfd-model-rif-samples/src/main/resources/rif-static-samples/sample-u-beneficiaries.txt:2]: Same as above: lean more towards non-realistic data.

File: apps/bfd-model/bfd-model-rif/src/main/java/gov/cms/bfd/model/rif/XmlNamespace.java
Comment (karlmdavis) [inline apps/bfd-model/bfd-model-rif/src/main/java/gov/cms/bfd/model/rif/XmlNamespace.java:6]: WTG!

File: apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V26__Add_fields_to_BeneficiariesInpatient.sql
Comment (karlmdavis) [inline apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V26__Add_fields_to_BeneficiariesInpatient.sql:26]: What do ACOs want to do with these two fields? If we only add them here (as opposed to also adding them to `"BeneficiariesHistory"` AND backfilling/updating the data for it) then we'll only have them for a bene's most recent/current MBI -- we won't be able to use or return them for historical MBIs.

File: apps/bfd-server-test-perf/src/test/resources/beneficiary_truncated.rif
Comment (karlmdavis) [inline apps/bfd-server-test-perf/src/test/resources/beneficiary_truncated.rif:2]: Same as above: lean more towards non-realistic data.

File: apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V26__Add_fields_to_BeneficiariesInpatient.sql
Comment (karlmdavis) [inline apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V26__Add_fields_to_BeneficiariesInpatient.sql:24]: Should this field be `not null`? It's not clear from the documentation you provided in the PR description.

File: apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V26__Add_fields_to_BeneficiariesInpatient.sql
Comment (karlmdavis) [inline apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V26__Add_fields_to_BeneficiariesInpatient.sql:22]: How does the data in this field differ from `"Beneficiaries"."postalCode"`?

File: apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V26__Add_fields_to_BeneficiariesInpatient.sql
Comment (karlmdavis) [inline apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V26__Add_fields_to_BeneficiariesInpatient.sql:20]: How does the data in this field differ from `"Beneficiaries"."stateCode"`?

**Review by karlmdavis — DISMISSED** [review]
Solid work!

I noted a few things that need to be changed and also left in several questions that should be addressed before proceeding.

File: apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V26__Add_fields_to_BeneficiariesInpatient.sql
Comment (cbrunefearless) [reply apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V26__Add_fields_to_BeneficiariesInpatient.sql:26]: Added to beneficiary history as well.

File: apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V26__Add_fields_to_BeneficiariesInpatient.sql
Comment (cbrunefearless) [reply apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V26__Add_fields_to_BeneficiariesInpatient.sql:22]: From GEDIT->. The following variables in the current FHIR extract are from the EDB source:

·        STATE_CD

·        CNTY_CD

·        BENE_ZIP

 

These new variables will be from the CME Derived Mailing source:

·        DRVD_LINE_1_ADR

·        DRVD_LINE_2_ADR

·        DRVD_LINE_3_ADR

·        DRVD_LINE_4_ADR

·        DRVD_LINE_5_ADR

·        DRVD_LINE_6_ADR

·        CITY_NAME

 

We are finding some differences in the State Code between the two sources.


So we did find differences which is why we proposed this.

File: apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V26__Add_fields_to_BeneficiariesInpatient.sql
Comment (cbrunefearless) [reply apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V26__Add_fields_to_BeneficiariesInpatient.sql:20]: answer above..

File: apps/bfd-model/bfd-model-rif-samples/src/main/resources/rif-static-samples/sample-a-beneficiaries.txt
Comment (cbrunefearless) [reply apps/bfd-model/bfd-model-rif-samples/src/main/resources/rif-static-samples/sample-a-beneficiaries.txt:2]: yes sir

File: apps/bfd-model/bfd-model-rif-samples/src/main/resources/rif-static-samples/sample-mct-update-1-beneficiaries.txt
Comment (cbrunefearless) [reply apps/bfd-model/bfd-model-rif-samples/src/main/resources/rif-static-samples/sample-mct-update-1-beneficiaries.txt]: Done

File: apps/bfd-model/bfd-model-rif-samples/src/main/resources/rif-static-samples/sample-mct-beneficiaries.txt
Comment (cbrunefearless) [reply apps/bfd-model/bfd-model-rif-samples/src/main/resources/rif-static-samples/sample-mct-beneficiaries.txt]: Done

File: apps/bfd-model/bfd-model-rif-samples/src/main/resources/rif-static-samples/sample-hicn-mult-bene-beneficiaries.txt
Comment (cbrunefearless) [reply apps/bfd-model/bfd-model-rif-samples/src/main/resources/rif-static-samples/sample-hicn-mult-bene-beneficiaries.txt]: Done

File: apps/bfd-server-test-perf/src/test/resources/beneficiary_truncated.rif
Comment (cbrunefearless) [reply apps/bfd-server-test-perf/src/test/resources/beneficiary_truncated.rif]: Done

File: apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V26__Add_fields_to_BeneficiariesInpatient.sql
Comment (cbrunefearless) [reply apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V26__Add_fields_to_BeneficiariesInpatient.sql]: All fields they said were allowed to be null

**Comment by JFU-GIT** [comment]
Hi Chri, pushed up changes for BFD-379, take a look when you got chance, thanks!

**Review by jzulim — APPROVED** [review]
This is looking great. Fields, Migrations, Ingestion and tests (updated sample data) looks good. Thanks for the peer review. Let it rip!

**Review by dshekhar18 — APPROVED** [review]
@cbrunefearless Please get the field descriptions written down somewhere in the Git repo, as these descriptions change.

**Pull request merged**
